lib/spack/spack/stage.py

```diff
@@ -488,11 +489,13 @@ def create(self):
         if self._need_to_create_path():
             tmp_root = get_tmp_root()
             if tmp_root is not None:
+                # tempfile.mkdtemp already sets mode 0700
                 tmp_dir = tempfile.mkdtemp('', _stage_prefix, tmp_root)
                 tty.debug('link %s -> %s' % (self.path, tmp_dir))
                 os.symlink(tmp_dir, self.path)
             else:
-                mkdirp(self.path)
+                # emulate file permissions for tempfile.mkdtemp
+                mkdirp(self.path, mode=stat.S_IRWXU)
```

Comment:
So, I find the IRWXU stuff really unreadable.  I know what it means, but I'm *so* used to the chmod syntax and I think other maintainers will be too.  What if we allow `mkdirp` to accept `[ugo][+-][rwxX]`?  I actually don't think it's hard to go from a string like that to a function that does the intended thing.  Would you be up for that?

Reply:
@tgamblin I really don't think it's worthwhile to do this. I found a github gist for symbolic chmod in python that doesn't support the values 'X', 's', or 't', at least two of which we need. It looks harder to maintain than anything we'll write using the IRWXU stuff.

I think we can get better maintainability by just writing comments with the equivalent symbolic modes when we do the IRWXU stuff.

Then again, I'm the weirdo who prefers octal modes to symbolic modes for chmod, so maybe I'm not the best one to answer whether this is worthwhile.

Reply:
Then just leave it for now. I don’t think it would be hard to maintain once we implement it if it had good tests, but for now things are probably good enough.